### PR TITLE
test: run fork test against mainnet deployment

### PR DESCRIPTION
**Pull request by smol-ninja**

### CODE REVIEW

File: tests/fork/Fork.t.sol [hunk 1]
```diff
@@ -37,11 +38,16 @@ abstract contract Fork_Test is Base_Test {
     //////////////////////////////////////////////////////////////////////////*/
 
     function setUp() public virtual override {
-        // Fork Ethereum Mainnet at a specific block number. The block number is for the `OCT_1_2024` date.
-        vm.createSelectFork({ blockNumber: 20_866_919, urlOrAlias: "mainnet" });
+        // Fork Ethereum Mainnet at a block number during the Sablier deployment.
```
Comment (andreivladbrg) [inline tests/fork/Fork.t.sol:41]: let's say "after": `Fork Ethereum Mainnet at a block number after the Sablier deployment.`

File: tests/fork/Fork.t.sol [hunk 2]
```diff
@@ -37,11 +38,16 @@ abstract contract Fork_Test is Base_Test {
...
+        vm.createSelectFork({ blockNumber: 21_330_577, urlOrAlias: "mainnet" });
 
-        // The base is set up after the fork is selected so that the base test contracts are deployed on the fork.
-        Base_Test.setUp();
+        // Load mainnet address of flow.
+        flow = SablierFlow(0x2D9221a63E12AA796619cb381Ec4A71b201281f5);
```
Comment (andreivladbrg) [inline tests/fork/Fork.t.sol:45]: let's use `ISablierFlow`:

```solidity
flow = ISablierFlow(0x2D9221a63E12AA796619cb381Ec4A71b201281f5);
```

File: tests/fork/Fork.t.sol [hunk 3]
```diff
@@ -37,11 +38,16 @@ abstract contract Fork_Test is Base_Test {
...
+
+        // Label the flow contract.
+        vm.label(address(flow), "Flow");
+
+        resetPrank(users.sender);
```
Comment (andreivladbrg) [inline tests/fork/Fork.t.sol:50]: why do we need to add this?

Comment (smol-ninja) [reply tests/fork/Fork.t.sol:50]: mb we do not. I will run a test without it.

File: tests/fork/Fork.t.sol (hunk 2 above)
Comment (smol-ninja) [reply tests/fork/Fork.t.sol:45]: Unfortunately, doing this throws error. https://sablier.slack.com/archives/C06BB90SY02/p1733579717082119

**Comment by smol-ninja** [comment]
https://github.com/sablier-labs/flow/pull/345/commits/21da5ac55037d65f9589cc86e13fc385c96222b1 includes all your suggestions @andreivladbrg.

**Review by andreivladbrg — APPROVED** [review]
lgtm_meme

**Pull request merged**